sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/TableClient.java

```diff
@@ -207,7 +365,40 @@ public void deleteTable() {
      */
     @ServiceMethod(returns = ReturnType.SINGLE)
     public Response<Void> deleteTableWithResponse(Duration timeout, Context context) {
-        return blockWithOptionalTimeout(client.deleteTableWithResponse(context), timeout);
+        // TODO: Spawn thread to utilize timeout param
+        Context contextValue = setContext(context);
+        final ScheduledExecutorService scheduler = Executors.newScheduledThreadPool(1);
+        long timeoutInMillis = setTimeout(timeout);
+
+        Callable<Response<Void>> deleteTableOp = () -> {
+            return new SimpleResponse<>(tablesImplementation.getTables().deleteWithResponse(
+                tableName, null, contextValue),
+                null);
+        };
+
+        ScheduledFuture<Response<Void>> scheduledFuture =
+            scheduler.schedule(deleteTableOp, IMMEDIATELY, TimeUnit.SECONDS);
+
+        scheduler.shutdown();
+        try {
+            return scheduledFuture.get(timeoutInMillis, TimeUnit.MILLISECONDS);
+        } catch (Exception ex) {
+            Throwable except = mapThrowableToTableServiceException(ex);
+            return swallow404Exception(except);
+        }
+    }
+
+    private Response<Void> swallow404Exception(Throwable ex) {
```

Comment:
Can we reuse the existing swallowExceptionForStatusCode from async client?

Reply:
@samvaity Nope, it returns a Mono.

Reply:
We should be able to return the exception and then wrap it with respective needs for Mono or otherwise based on the call stack for sync/async calls.
Similar example - https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/containerregistry/azure-containers-containerregistry/src/main/java/com/azure/containers/containerregistry/ContainerRegistryAsyncClient.java#L181

This will help ensure code reusability and less checks when updating common functionality.

Reply:
are you planning to take the refactoring in a separate PR?